src/modules/directory.rs

```diff
@@ -149,12 +147,12 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
                 _ => None,
             })
             .map(|variable| match variable {
-                "path" => Some(Ok(&path_vec[2])),
-                "before_root_path" => Some(Ok(&path_vec[0])),
-                "repo_root" => Some(Ok(&path_vec[1])),
+                "path" => Some(Ok(path_vec[2].as_str())),
+                "before_root_path" => Some(Ok(path_vec[0].as_str())),
+                "repo_root" => Some(Ok(path_vec[1].as_str())),
                 "read_only" => {
                     if is_readonly_dir(physical_dir) {
-                        Some(Ok(&lock_symbol))
+                        Some(Ok(config.read_only))
```

Comment:
Shouldn't `as_str` usually match `&`?

Reply:
No, since `&` converts into `&String` (not `&str`), which `config.read_only` would not fit.